fedimint-core/src/core.rs

```diff
@@ -322,15 +343,53 @@ impl Decoder {
         DecoderBuilder::default()
     }
 
+    /// System Dyn-type, don't use.
+    #[doc(hidden)]
+    pub fn builder_system() -> DecoderBuilder {
```

Comment:
A bit of a ... workaround... . `DynState` is not really a `DynType` in the same sense as `DynInput` etc. are - it doesn't belong to a real module. But it does work exactly the same, so it's convenient to just make up and ID etc. for it. But architecturally it's a hack, and it's going to lead to papercuts and workarounds like here.

Reply:
I think we could get rid of nested dyn types by serializing the transaction before putting it into the tx state machine. Over the wire it's sent as hex-encoded bytes anyway (using `SerdeModuleEncoding<Transaction>`).

Reply:
Might require migration, possibly extra work -> outside of scope of this change. :)